cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -37,12 +38,15 @@ const pathSeparator = "\x1F"
 
 var storageReferenceEncodingStart = []byte{0xd8, cborTagStorageReference}
 
-var storageMigrationV6DecMode = func() cbor.DecMode {
+// maxInt is math.MaxInt32 or math.MaxInt64 depending on arch.
+const maxInt = 1<<(bits.UintSize-1) - 1
+
+var storageMigrationV5DecMode = func() cbor.DecMode {
 	decMode, err := cbor.DecOptions{
 		IntDec:           cbor.IntDecConvertNone,
-		MaxArrayElements: math.MaxInt32,
```

Comment:
Are we actually running into these limits here?

Reply:
Not really, I just updated it to be on sync with decode-mode defined in format-V5